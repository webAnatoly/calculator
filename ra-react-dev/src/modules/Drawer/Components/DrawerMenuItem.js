import React, { Component } from 'react'
import { ListItem, ListItemIcon, ListItemText } from 'material-ui/List'

export default class DrawerMenuItem extends Component {
  render () {
    const { label, onClick, Icon } = this.props
    return (
      <ListItem button onClick={onClick}>
        <ListItemIcon>
          <Icon />
        </ListItemIcon>
        <ListItemText primary={label} />
      </ListItem>
    )
  }
}
