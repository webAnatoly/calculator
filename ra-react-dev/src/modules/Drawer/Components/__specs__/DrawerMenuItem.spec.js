import React from 'react'
import DrawerMenuItem from '../DrawerMenuItem'
import { shallow } from 'enzyme'

describe('DrawerMenuItem', () => {
  let wrapper, onClick, label, Icon
  beforeEach(() => {
    label = 'test'
    onClick = jest.fn()
    Icon = jest.fn()
    wrapper = shallow(
      <DrawerMenuItem label={label} onClick={onClick} Icon={Icon} />
    )
  })

  it('should renders as expected', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should handle onClick()', () => {
    wrapper.props().onClick()
    expect(onClick.mock.calls.length).toEqual(1)
  })

  it('should render ListItemText with label', () => {
    expect(wrapper.childAt(1).props().primary).toEqual(label)
  })
})
