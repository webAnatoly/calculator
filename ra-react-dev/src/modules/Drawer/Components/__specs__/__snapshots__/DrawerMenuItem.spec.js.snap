// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerMenuItem should renders as expected 1`] = `
<withStyles(ListItem)
  button={true}
  onClick={[Function]}
>
  <withStyles(ListItemIcon)>
    <mockConstructor />
  </withStyles(ListItemIcon)>
  <withStyles(ListItemText)
    primary="test"
  />
</withStyles(ListItem)>
`;
